src/test/run-pass/dynamic-drop.rs

```diff
@@ -111,6 +111,20 @@ fn assignment1(a: &Allocator, c0: bool) {
     _v = _w;
 }
 
+#[allow(unions_with_drop_fields)]
+union Boxy<T> {
+    a: T,
+    b: T,
+}
+
+fn union1(a: &Allocator) {
+    unsafe {
+        let mut u = Boxy { a: a.alloc() };
+        u.b = a.alloc();
+        drop(u.a);
```

Comment:
I too was worried about whether this scenario is well-defined in general, or if it is potentially undefined behavior.

According to the union RFC that I see here: https://github.com/rust-lang/rfcs/blob/master/text/1444-union.md#unions-and-undefined-behavior, you cannot read from a field other than the one written to unless you have provided extra info in the `repr` attribute ensuring that the compiler is actually picking a suitable layout.

Since one cannot read from `u.b` at the point where it is being overwritten here, it seems logical to me that one should also infer that one cannot drop `u.b` ...

So, I suggest (in the spirit of the linked section of the `union` RFC) that you revise the test so that `struct Boxy` has a `#[repr(C)]` on it.

(However, I anticipate push-back from e.g. @nikomatsakis on this point. ;) )